scene/2d/camera_2d.cpp

```diff
@@ -409,39 +421,34 @@ void Camera2D::set_current(bool p_current) {
 	if (p_current) {
 		make_current();
 	} else {
-		if (current) {
-			clear_current();
-		}
+		clear_current();
```

Comment:
Isn't the point of this that if `current` is false, there's no need for `clear_current()` to `call_group` etc?

Reply:
Okay. I've update the PR with a refactored `set_current()` to make things simpler and clearer.
Basically, when calling `set_current(true)` it now calls `_clear_current()` on all cameras in the `Viewport`, which just sets `current` to `false` and `redraws` it. It then sets this camera to be current. When calling `set_current(false)` it just sets this camera to not be current.